pkg/snmp/traps/formatter.go

```diff
@@ -124,48 +130,62 @@ func (f JSONFormatter) formatV1Trap(packet *gosnmp.SnmpPacket) map[string]interf
 	data["enterpriseOID"] = enterpriseOid
 	data["genericTrap"] = genericTrap
 	data["specificTrap"] = specificTrap
-	parsedVariables, enrichedValues := f.parseVariables(trapOID, packet.Variables)
+	parsedVariables, enrichedValues := f.parseVariables(trapOID, content.Variables)
+	enrichmentFailed := len(content.Variables) - len(enrichedValues)
+	if enrichmentFailed > 0 {
+		f.aggregator.Count("datadog.snmp_traps.vars_not_enriched", float64(enrichmentFailed), "", tags)
+	}
 	data["variables"] = parsedVariables
 	for key, value := range enrichedValues {
 		data[key] = value
 	}
 	return data
 }
 
-func (f JSONFormatter) formatTrap(packet *gosnmp.SnmpPacket) (map[string]interface{}, error) {
+func (f JSONFormatter) formatTrap(packet *SnmpPacket) (map[string]interface{}, error) {
 	/*
 		An SNMP v2 or v3 trap packet consists in the following variables (PDUs):
 		{sysUpTime.0, snmpTrapOID.0, additionalDataVariables...}
 		See: https://tools.ietf.org/html/rfc3416#section-4.2.6
 	*/
-	variables := packet.Variables
+	tags := f.getTags(packet)
+
+	variables := packet.Content.Variables
 	if len(variables) < 2 {
+		f.aggregator.Count("datadog.snmp_traps.incorrect_format", 1, "", append(tags, "error:invalid_variables"))
```

Comment:
Especially as we use a counter that's being reused in different places it might be worth to use a constant for the metric name

Reply:
On it!